src/main/kotlin/interactor/Consumption.kt

```diff
@@ -0,0 +1,62 @@
+package interactor
+
+import model.CityEntity
+
+
+class Consumption (private val dataSource: CostOfLivingDataSource){
+
+    fun execute(): String {
+         val x:MutableMap<Float,MutableList<String>> = mutableMapOf(Pair((0).toFloat(), mutableListOf()))
+         val cities=dataSource
+            .getAllCitiesData()
+            .filter(::excludeNullValue)
+            .sortedBy{it.realEstatesPrices.apartment3BedroomsOutsideOfCentre}
+        cities.forEach {
+             val servicesCoast= servicesCoast(it)
+             val remainder= remainder(it,servicesCoast)
+             if(isSuitable(servicesCoast,remainder)){
+                 if (x.contains(remainder)) x[remainder]?.add(it.cityName)
+                 else x[remainder] = mutableListOf(it.cityName)
+             }
+         }
+        return x.toSortedMap()
+            .values
+            .last()
+            .toString()
+    }
+}
+private fun excludeNullValue(city: CityEntity): Boolean {
+    return city.averageMonthlyNetSalaryAfterTax != null
+            && city.dataQuality
```

Comment:
He didn't mention to check the dataQuality in task 10, so you shouldn't exclude values according to dataQuality